htmd/smallmol/smallmol.py

```diff
@@ -1,105 +1,654 @@
+    >>> sm = SmallMol( 'htmd/data/test-smallmol/ligand.pdb', fixHs=False, removeHs=True )
+    >>> sm = SmallMol( 'htmd/data/test-smallmol/benzamidine.mol2')
+    >>> print(mol)                                     # doctest: +ELLIPSIS
+    SmallMol with 18 atoms and
+    Atom field - atomname shape: (18,)
+    Atom field - bondtypes shape: (18,)
+    ...
+
+    .. rubric:: Methods
+    .. autoautosummary:: htmd.smallmol.smallmol.SmallMol
+       :methods:
+    .. rubric:: Attributes
+    .. autoautosummary:: htmd.smallmol.smallmol.SmallMol
+       :attributes:
+
+    Attributes
+    ----------
+
+    listProps : list
+        List of the molecule property
+    listPropsAtoms: list
+        List of the atom property
+    numConformers: int
+        Number of confomrmers
+    numAtoms: int
+        Number of atoms
+
     """
-    array_cache = {(16, 1.): _getGridCenters(np.array([-8] * 3),
-                                               [16, 16, 16], 1.).reshape(16**3, 3),
-                   (24, 1.): _getGridCenters(np.array([-12] * 3),
-                                               [24, 24, 24], 1.).reshape(24**3, 3)}
 
-    fdefName = os.path.join(RDConfig.RDDataDir, 'BaseFeatures.fdef')
-    factory = ChemicalFeatures.BuildFeatureFactory(fdefName)
+    _atom_fields = ['idx', 'atomname', 'charge','formalcharge', 'element',  'chiral', 'hybridization',
+                    'neighbors', 'bondtypes', 'coords', '_chiraltags']
+
+    _mol_fields = ['ligname', 'totalcharge', '_mol']
+
+    ### Field types
+    ## Development Note:  'expliciths' removed.
+    _atom_dtypes = {'idx': np.int,
+                    'atomname': object,
+                    'charge': np.float32,
+                    'formalcharge': np.int,
+                    'element': object,
+                    'chiral': object,
+                    'hybridization': int,
+                    'neighbors': object,
+                    'bondtypes': int,
+                    'coords': np.float32,
+                    '_chiraltags': object
+                    }
+
+    _mol_dtypes = {'_mol': object,
+                    'ligname': object,
+                   'totalcharge': np.int
+                   }
+
+    ### Field shapes
+    ## Development Note:  'expliciths' removed.
+    _atom_dims = {'idx': (0,),
+                  'atomname': (0,),
+                  'charge': (0,),
+                  'formalcharge': (0,),
+                  'element': (0,),
+                  'chiral': (0,),
+                  'neighbors': (0,),
+                  'bondtypes': (0,),
+                  'hybridization': (0,),
+                  'coords': (0, 3, 1),
+                  '_chiraltags':(0,)
+                    }
+
+    _mol_dims = {'_mol': (0,),
+                 'ligname': (0,),
+                 'totalcharge': (0,)
+                   }
+
+    def __init__(self, mol, ignore_errors=False, force_reading=False, fixHs=True, removeHs=False):
+
+
+        self.mol_fields = self._mol_fields.copy()
```

Comment:
No need to copy class static variables unless you plan on modifying them (which I don't see you doing anywhere). If you want to make them public just remove the underscore.

Reply:
_chiraltags is modified in chirals compounds. Due to the fact that is not a chemist readable value I decided to hide it with the underscore. I mean that is necessary to copy it because it would be modified, but I also would like to keep it hidden. If you have a better way to do it I will modify it.

Regarding the '_mol'. This is the origianl rdkit mol. I keep it for debugging purpose only. Here the reason why it is hidden with the underscore. Can I keep it like that or do you suggest a better way?

Reply:
But `_mol_fields = ['ligname', 'totalcharge', '_mol']`. This is a list of strings. You are copying a list of strings. Do you mean that you append to this list of strings somewhere a new string?

Reply:
Yes. This list of strings was created to store any kind of molecule property. For instance, the SmallMolLib use this list to store the sdf fields.

Reply:
I see okay thanks! :)